src/react-hook/react-hook.tsx

```diff
@@ -19,12 +20,19 @@ export function useConfig() {
       "ModuleNameContext has not been provided. This should come from openmrs-react-root-decorator"
     );
   }
+  if (error) {
+    throw error;
+  }
   if (!config) {
     // React will prevent the client component from rendering until the promise resolves
-    throw Config.getConfig(moduleName).then(res => {
-      config = res;
-      //      setConfig(res);
-    });
+    throw Config.getConfig(moduleName)
+      .then(res => {
+        config = res;
+        //      setConfig(res);
+      })
+      .catch(err => {
+        error = err;
```

Comment:
This is not a good way to handle errors in React. Calling `.catch()` will make a new promise that is in "fulfilled" status (even though it started out in "rejected" status), which means that suspense will finish "successfully." When react sees that suspense has successfully finished, it triggers a re-render. That re-render is what was causing the infinite loop, since it would result in another call to `useConfig()` and another `throw Config.getConfig().catch()`.

I do not think we should try to store the error and avoid calling `Config.getConfig()` if it's broken. I think it's okay for a unique error to be created for each call to `useConfig`. Here's what we should do instead:

```tsx
throw Config.getConfig(moduleName)
  .then(res => {
    config = res
  })
  .catch(err => {
    // setTimeout throw will ensure this makes it out to the global error handler instead of just this promise. The global error handler prints it to the console and shows an error toast/growl thing so the dev and user know that something happened
    setTimeout(() => {
      throw err
    })

    // throwing without setTimeout changes our promise back into `rejected` status (which is good since we don't want suspense to think that this promise worked!
    throw err;
  })
```

^ This will ensure that the error makes it to the console.

Reply:
Also check out [this codepen](https://codepen.io/joeldenning/pen/wvvQjrg?editors=0010) and [this tweet](https://twitter.com/Joelbdenning/status/1195533397868679168?s=20). I initially thought that when the "suspense promise" rejects that it would trigger [this error boundary](https://github.com/openmrs/openmrs-react-root-decorator/blob/master/src/openmrs-react-root-decorator.js#L62). But it didn't in my testing in the codepen, which disappointed me because that would have been a good way to handle this error.

Reply:
I replaced my code with this approach, and the infinite looping behavior returned. [Screenshot](https://ibb.co/zQWxhmT).

The throw in setTimeout is succeeding in printing the error to the console, which is cool! That's a big improvement over the previous silent-infinite-loop behavior.

See [your pen with a console.log statement](https://codepen.io/bistenes/pen/XWWoovx?editors=0011), or [this simplified pen](https://codepen.io/bistenes/pen/VwwqqNZ).

Reply:
The reason is that my codepen didn't render suspense inside of an error boundary. See [this codepen that works](https://codepen.io/joeldenning/pen/wvvQjrg?editors=0010) and this [tweet from Dan Abramov](https://twitter.com/dan_abramov/status/1195656371875860481?s=20) that explained it to me [with his own codepen](https://codesandbox.io/s/adoring-goodall-8wbn7).